.github/workflows/ci.yml

```diff
@@ -66,6 +66,37 @@ jobs:
           node-version: ${{ matrix.node }}
       - run: corepack yarn
       - run: corepack yarn vitest run --coverage ./test/unit
+      - name: Upload coverage reports
+        if: matrix.node == 22 # Only upload coverage from the latest Node.js version
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-reports
+          path: coverage/
+      - name: Upload coverage to Codecov
+        if: matrix.node == 22 # Only upload coverage from the latest Node.js version
```

Comment:
maybe we can move this into the `coverage` job, then we don't have to check for node version again also